### PR TITLE
Add type subtype item association model

### PR DESCRIPTION
**Pull request by kti-cameron**

### CODE REVIEW

File: djconnectwise/models.py [hunk 1]
```diff
@@ -2126,6 +2129,20 @@ def __str__(self):
...
 
+class TypeSubTypeItemAssociation(TimeStampedModel):
+    type = models.ForeignKey('Type', blank=True, null=True,
+                             related_name='associations',
+                             on_delete=models.CASCADE)
+    sub_type = models.ForeignKey('SubType', blank=True,
+                                 related_name='associations',
+                                 null=True, on_delete=models.CASCADE)
+    item = models.ForeignKey('Item', blank=True,
+                             related_name='associations',
```
Comment (kti-daniel) [inline djconnectwise/models.py:2140]: Question: Can `related_name`s have the same name?

File: djconnectwise/sync.py [hunk 2]
```diff
@@ -93,6 +93,7 @@ class Synchronizer:
 
     def __init__(self, full=False, *args, **kwargs):
         self.api_conditions = []
+        self.partial_sync_support = True
```
Comment (kti-daniel) [inline djconnectwise/sync.py:96]: Can I ask for your explanation for an introduction of `partial_sync_support`? Until now, we support partial & full sync wo/ any other value, so why is that added here?

File: djconnectwise/models.py (hunk 1 above)
Comment (kti-cameron) [reply djconnectwise/models.py:2140]: In this case I believe the answer is yes. 

The default `related_name` if I hadn't changed it to `associations` would be `typesubtypeitemassociation_set` and that would have been the same for `type`, `sub_type` and `item`. 

For example:
```
In : st = SubType.objects.first()
In : st
<SubType: Scanner>
In : st.typesubtypeitemassociation_set.all()
<QuerySet []>
In : t.typesubtypeitemassociation_set.all()
<QuerySet []>
In : t
<Type: Break-fix - Professional Services>
In : 
```
My understanding is that the related_name would have to be unique if our model had more than one FK relationship to another model. For example, we would have problems if the model looked like:
```
class TypeSubTypeItemAssociation(TimeStampedModel):
    type_1 = models.ForeignKey('Type'')
    type_2 = models.ForeignKey('Type')
```
type_1 and type_2 would need to have unique related_name's added to differentiate themselves from each other.

File: djconnectwise/sync.py (hunk 2 above)
Comment (kti-cameron) [reply djconnectwise/sync.py:96]: I had to add this since the `typeSubTypeItemAssociations` endpoint doesn't have  a `lastUpdated` field. We need to know when records were last updated in order to perform a partial sync. So if the `lastUpdated` field is not present we can only ask the API for all records.

File: djconnectwise/models.py (hunk 1 above)
Comment (kti-daniel) [reply djconnectwise/models.py:2140]: Thanks for the explanation

File: djconnectwise/sync.py (hunk 2 above)
Comment (kti-daniel) [reply djconnectwise/sync.py:96]: Can I ask one thing here?

Work_role also doesn't have `lastUpdated` field. I don't look into that deeply, but doesn't this time info be used to compare the info in our sync log job table?

image

Comment (kti-cameron) [reply djconnectwise/sync.py:96]: I did expect `lastUpdated` to be available from "info" but it is just not there.
image

My guess is that since these records can't be changed directly by users the updatedBy and lastUpdated fields are not available. 

This is what I got when trying to query with lastUpdated.
image

Comment (kti-daniel) [reply djconnectwise/sync.py:96]: I got it. Thanks for the clarifications.
